app/uk/gov/hmrc/tradergoodsprofiles/controllers/RemoveRecordController.scala

```diff
@@ -55,6 +55,6 @@ class RemoveRecordController @Inject() (
   ToDo: remove this validation for drop2 - TGP-2029
    */
   private def validateHeadersForDrop2(implicit request: Request[_]): Either[Result, _] =
-    if (appConfig.isDrop2Enabled) Right("Success")
+    if (appConfig.isDrop2Enabled || appConfig.acceptHeaderEnabled) Right("Success")
```

Comment:
The logic here is not quite right. If the isDrop2Enabled is true this will validate the accept header no matter what the value of acceptHeaderEnabled is and viceversa.

```suggestion
    (for {
      _ <- if(!appConfig.isDrop2Enabled) validateAcceptHeader else Right(Done)
      _ <- if(!appConfig.acceptHeaderEnabled) validateClientIdHeader else Right(Done)
    } yield ()).left
      .map(e => createBadRequestResponse(e.code, e.message, e.errorNumber))
```

Also I would validate the accept Header if acceptHeaderEnabled is true,  I think this would be more readable